sdk/eventhub/azure-eventhub-checkpointstoreblob/azure/eventhub/extensions/checkpointstoreblob/_blobstoragecs.py

```diff
@@ -136,9 +136,13 @@ def _upload_ownership(self, ownership, metadata):
             ownership["partition_id"],
         )
         blob_name = blob_name.lower()
-        uploaded_blob_properties = self._get_blob_client(blob_name).upload_blob(
-            data=UPLOAD_DATA, overwrite=True, metadata=metadata, **etag_match
-        )
+        blob_client = self._get_blob_client(blob_name)
+        try:
+            uploaded_blob_properties = blob_client.set_blob_metadata(metadata, **etag_match)
+        except ResourceNotFoundError:
```

Comment:
Lets add a logging debug stanza in each of these forks.  I can foresee this being the root of weirdness down the road (I don't EXPECT problems, it's just...  paranoia :P) and then I can rubber stamp this.